### ISSUE TITLE
Solve for polynomial equation

### ISSUE THREAD

**Issue by mikldk** [grzegorzmazur/yacas#278]
This is related to #272. I just tried the finding critical points in a Rosenbrock function:

The function is `(1 - x)^2 + 100*(y - x^2)^2`. The partial derivative wrt. `x` is `(-2)*(1-x)-400*x*(y-x^2)` and wrt. `y` it is `200*(y-x^2)`. Trying to find roots gives

```
Solve({(-2)*(1-x)-400*x*(y-x^2), 200*(y-x^2)}, {x, y})
# Wrong Groebner basis
```

I am not sure whether this is because Groebner bases cannot solve this problem or whether the implementation is missing something.

[Wolfram Alpha](https://www.wolframalpha.com/input/?i=solve+%28-2%29*%281-x%29-400*x*%28y-x%5E2%29+%3D%3D+0%2C+200*%28y-x%5E2%29+%3D%3D+0+for+x%2C+y) finds the solution `(x, y) = (1, 1)`.

**Comment by grzegorzmazur** [comment]
Hi,

Good catch! The message is all wrong, the correct Groebner basis is generated, only it just happens to break an assumption I've made in solver. Seems fairly easy to fix, but if there's one wrong assumption there may be as well some other, so I'll think about it a couple of days before commiting a fix. 

Thanks,
Grzesiek

**Comment by mikldk** [comment]
Thanks. It's not at all urgent. I just reported my observation because I hoped it could help improve yacas.

Should this and the other system be included as test cases?

Thanks!

**Comment by grzegorzmazur** [comment]
Definitely! I've already added this system as a testcase (I try to do it for every fix). The ones you've reported previously were added along with the code for solving them.

And thanks a lot for the reports, it's much easier to fix issues I'm aware of :)

**Comment by mikldk** [comment]
Thanks!
